### PR TITLE
fix calc_sever print_status npe bug

### PR DESCRIPTION
**Pull request by tianjian600526**
1、 fix ssl library path not found in my mac m1.
2、fix example npe bugs.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -85,8 +85,14 @@ if (NOT WIN32)
...
-        include_directories(BEFORE /usr/local/opt/openssl/include)
-        link_directories(/usr/local/opt/openssl/lib)
+        include_directories(BEFORE
+                /usr/local/opt/openssl/include
+                /opt/homebrew/opt/openssl@1.1/include
+                )
+        link_directories(
+                /usr/local/opt/openssl/lib
+                /opt/homebrew/opt/openssl@1.1/lib
+                )
```
Comment (greensky00) [inline CMakeLists.txt:95]: Can you please make a separate PR for this? There are two different fixes here. And also I recommend
```
if (APPLE)
    execute_process(COMMAND brew --prefix OUTPUT_VARIABLE BREW_PREFIX)
    message(STATUS "brew prefix ${BREW_PREFIX}")
    include_directories(BEFORE
        ${BREW_PREFIX}/opt/openssl@1.1/include
        /usr/local/opt/openssl/include
...
```

File: examples/calculator/calc_server.cxx [hunk 2]
```diff
@@ -152,9 +152,9 @@ void print_status(const std::string& cmd,
         << "current term: "
             << stuff.raft_instance_->get_term() << std::endl
         << "last snapshot log index: "
-            << stuff.sm_->last_snapshot()->get_last_log_idx() << std::endl
+            << ((stuff.sm_->last_snapshot() != nullptr) ? stuff.sm_->last_snapshot()->get_last_log_idx() : 0) << std::endl
         << "last snapshot log term: "
-            << stuff.sm_->last_snapshot()->get_last_log_term() << std::endl
+            << ((stuff.sm_->last_snapshot() != nullptr) ? stuff.sm_->last_snapshot()->get_last_log_term() : 0) << std::endl
```
Comment (greensky00) [inline examples/calculator/calc_server.cxx:157]: Please break lines at 90 chars
https://github.com/eBay/NuRaft/blob/71c33400a53aa23055733958c41f2461d3d07b49/.clang-format#L7
Also it can be simplified as follows
```c++
(stuff.sm_->last_snapshot() ? stuff.sm_->last_snapshot()->get_last_log_idx() : 0)
```

File: CMakeLists.txt (hunk 1 above)
Comment (tianjian600526) [reply CMakeLists.txt:95]: You are right, I will do it later.

File: examples/calculator/calc_server.cxx (hunk 2 above)
Comment (tianjian600526) [reply examples/calculator/calc_server.cxx:157]: Okay.

**Review by greensky00 — APPROVED** [review]
Thanks!
